client/extensions/woocommerce/state/action-types.js

```diff
@@ -14,6 +14,9 @@ export const WOOCOMMERCE_PRODUCT_CREATE_SUCCESS = 'WOOCOMMERCE_PRODUCT_CREATE_SU
 export const WOOCOMMERCE_PRODUCT_EDIT = 'WOOCOMMERCE_PRODUCT_EDIT';
 export const WOOCOMMERCE_PRODUCT_UPDATED = 'WOOCOMMERCE_PRODUCT_UPDATED';
 export const WOOCOMMERCE_PRODUCT_VARIATION_EDIT = 'WOOCOMMERCE_PRODUCT_VARIATION_EDIT';
+export const WOOCOMMERCE_REQUEST = 'WOOCOMMERCE_REQUEST';
```

Comment:
Hmm, this interface would be abstract enough to work outside of jetpack though. (e.g. one day when this code might run outside of wpcom) so I'm not sure I like injecting JETPACK where it may not always belong.

How about `WOOCOMMERCE_RESTAPI_REQUEST`? That's probably descriptive enough. Do you think any of the other names need to change? Like `state/wc-api-request` instead of just `state/request`?